lucille-core/src/test/java/com/kmwllc/lucille/stage/ConfigValidationTest.java

```diff
@@ -57,33 +58,80 @@ public void testNonValidationModeException() throws Exception {
     assertFalse(result.getStatus());
   }
 
+  // asserts that if two connectors use the same pipeline it is only validated once
   @Test
-  public void testValidationModeException() throws Exception {
-    Map<String, List<Exception>> exceptions = Runner.runInValidationMode(addPath("pipeline.conf"));
-    assertEquals(2, exceptions.size());
+  public void testDuplicatePipeline() throws Exception {
+    Map<String, List<Exception>> exceptions = Runner.runInValidationMode(addPath("duplicate-pipeline.conf"));
+    assertEquals(1, exceptions.size());
 
-    List<Exception> exceptions1 = exceptions.get("connector1");
+    List<Exception> exceptions1 = exceptions.get("pipeline1");
     assertEquals(2, exceptions1.size());
 
-    List<Exception> exceptions2 = exceptions.get("connector2");
-    assertEquals(2, exceptions2.size());
+    testException(exceptions1.get(0), IllegalArgumentException.class,
+        "com.kmwllc.lucille.stage.NoopStage: " + "Stage config contains unknown property invalid_property");
+    testException(exceptions1.get(1), IllegalArgumentException.class, "Stage config must contain property fields");
+  }
 
-    testException(exceptions1.get(0), IllegalArgumentException.class, "com.kmwllc.lucille.stage.NoopStage: " +
-        "Stage config contains unknown property invalid_property");
+  // asserts that if no connectors use a pipeline it is still validated
+  @Test
+  public void testUnusedPipeline() throws Exception {
+    Map<String, List<Exception>> exceptions = Runner.runInValidationMode(addPath("no-used-pipelines.conf"));
+    assertEquals(1, exceptions.size());
+
+    List<Exception> exceptions1 = exceptions.get("pipeline1");
+    assertEquals(2, exceptions1.size());
 
-    // TODO note that for the following two exceptions, the fields are retrieved before
-    //  the config validation is called
+    testException(exceptions1.get(0), IllegalArgumentException.class,
+        "com.kmwllc.lucille.stage.NoopStage: " + "Stage config contains unknown property invalid_property");
 
-    testException(exceptions1.get(1), IllegalArgumentException.class,
-        "Stage config must contain property fields");
+    testException(exceptions1.get(1), IllegalArgumentException.class, "Stage config must contain property fields");
+  }
 
-    testException(exceptions2.get(0), IllegalArgumentException.class,
-        "Stage config must contain property dest");
+  @Test
+  public void testStringifyValidationExceptions() {
```

Comment:
Runner.validatePipelines uses a LinkedHashMap instead of a plain HashMap so key entry order is preserved. When we stringify the exception map, we should respect that order, and tests should confirm this. The goal is that the user running the validation command should see the output for each pipeline in the same order the pipelines occur in the config file. e.g. if the config has:

pipelines: [{name: "pipeline1", stages: [{}]}, {name: "pipeline2", stages: [{}]}]

we should see the validation output for pipeline1, then pipeline2